webpack/assets/javascripts/services/statisticsChartService.js

```diff
@@ -21,6 +22,37 @@ function getChartConfig(details) {
   return config;
 }
 
+function getChartConfigByData(_data, id, type) {
```

Comment:
the server should not know about "right format", as the desired format can change if, say, we change from c3 to other charting library. 
it should just send the data in a reasonable way.